File: clients/rust/src/models/rate_limiter_check_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::rate_limit_result::RateLimitResult;

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct RateLimiterCheckOut {
    /// Number of tokens remaining
    pub remaining: u64,

    /// Whether the request is allowed
    pub result: RateLimitResult,

    /// Seconds until enough tokens are available (only present when allowed is false)
    #[serde(skip_serializing_if = "Option::is_none")]
    pub retry_after: Option<u64>,
}

impl RateLimiterCheckOut {
    pub fn new(remaining: u64, result: RateLimitResult) -> Self {
        Self {
            remaining,
            result,
            retry_after: None,
        }
    }
}

File: clients/javascript/src/models/rateLimiterCheckOut.ts
// this file is @generated
import {
    type RateLimitStatus,
    RateLimitStatusSerializer,
} from './rateLimitStatus';

export interface RateLimiterCheckOut {
    /** Whether the request is allowed */
    status: RateLimitStatus;
    /** Number of tokens remaining */
    remaining: number;
    /** Seconds until enough tokens are available (only present when allowed is false) */
    retryAfter?: number | null;
}

export const RateLimiterCheckOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): RateLimiterCheckOut {
        return {
            status: RateLimitStatusSerializer._fromJsonObject(object['status']),
            remaining: object['remaining'],
            retryAfter: object['retry_after'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: RateLimiterCheckOut): any {
        return {
            'status': RateLimitStatusSerializer._toJsonObject(self.status),
            'remaining': self.remaining,
            'retry_after': self.retryAfter,
        };
    }
}